### PR TITLE
[ENG-522]Update testcases credentials, create reusable schema,

### PR DESCRIPTION
**Pull request by jlimatampersand**
This PR updates salesforce test cases to implement credential registry. 

Salesforce Test

```
jaehyunlim@Jaehyuns-MacBook-Pro connectors % go run test/salesforce/bulkwrite/main.go
Testing Bulkwrite...

=============================================
Testing Bulkwrite
=============================================

Upload complete.
{
    "state": "UploadComplete",
    "jobId": "750Dp0000090X8WIAU"
}
Write Result
{
    "id": "750Dp0000090X8WIAU",
    "object": "Touchpoint__c",
    "createdById": "005Dp000003Cd1SIAS",
    "externalIdFieldName": "external_id__c",
    "state": "Failed",
    "operation": "upsert",
    "columnDelimiter": "COMMA",
    "lineEnding": "LF",
    "numberRecordsFailed": 0,
    "numberRecordsProcessed": 0,
    "apexProcessingTime": 0,
    "apiActiveProcessingTime": 0,
    "apiVersion": 59,
    "concurrencyMode": "Parallel",
    "contentType": "CSV",
    "createdDate": "2024-03-19T23:30:28.000+0000",
    "jobType": "V2Ingest",
    "retries": 0,
    "systemModstamp": "2024-03-19T23:30:29.000+0000",
    "totalProcessingTime": 0
}


=============================================
Testing SuccessResults
=============================================

Upload complete.
{
    "state": "UploadComplete",
    "jobId": "750Dp0000090X8lIAE"
}
Write Result
{
    "jobId": "750Dp0000090X8lIAE",
    "state": "Failed",
    "jobInfo": {
        "id": "750Dp0000090X8lIAE",
        "object": "Touchpoint__c",
        "createdById": "005Dp000003Cd1SIAS",
        "externalIdFieldName": "external_id__c",
        "state": "Failed",
        "operation": "upsert",
        "columnDelimiter": "COMMA",
        "lineEnding": "LF",
        "numberRecordsFailed": 0,
        "numberRecordsProcessed": 0,
        "apexProcessingTime": 0,
        "apiActiveProcessingTime": 0,
        "apiVersion": 59,
        "concurrencyMode": "Parallel",
        "contentType": "CSV",
        "createdDate": "2024-03-19T23:30:28.000+0000",
        "jobType": "V2Ingest",
        "retries": 0,
        "systemModstamp": "2024-03-19T23:30:29.000+0000",
        "totalProcessingTime": 0
    },
    "message": "No records processed successfully. This is likely due the CSV being empty or issues with CSV column names."
}


=============================================
Testing Partial Failure
=============================================

Upload complete.
{
    "state": "UploadComplete",
    "jobId": "750Dp0000090X8gIAE"
}
Write Result
{
    "jobId": "750Dp0000090X8gIAE",
    "state": "Failed",
    "jobInfo": {
        "id": "750Dp0000090X8gIAE",
        "object": "Touchpoint__c",
        "createdById": "005Dp000003Cd1SIAS",
        "externalIdFieldName": "external_id__c",
        "state": "Failed",
        "operation": "upsert",
        "columnDelimiter": "COMMA",
        "lineEnding": "LF",
        "numberRecordsFailed": 0,
        "numberRecordsProcessed": 0,
        "apexProcessingTime": 0,
        "apiActiveProcessingTime": 0,
        "apiVersion": 59,
        "concurrencyMode": "Parallel",
        "contentType": "CSV",
        "createdDate": "2024-03-19T23:30:28.000+0000",
        "jobType": "V2Ingest",
        "retries": 0,
        "systemModstamp": "2024-03-19T23:30:29.000+0000",
        "totalProcessingTime": 0
    },
    "message": "No records processed successfully. This is likely due the CSV being empty or issues with CSV column names."
}


=============================================
Testing Complete Failure
=============================================

Upload complete.
{
    "state": "UploadComplete",
    "jobId": "750Dp0000090X8bIAE"
}
Write Result
{
    "jobId": "750Dp0000090X8bIAE",
    "state": "Failed",
    "jobInfo": {
        "id": "750Dp0000090X8bIAE",
        "object": "Touchpoint__c",
        "createdById": "005Dp000003Cd1SIAS",
        "externalIdFieldName": "external_id__c",
        "state": "Failed",
        "operation": "upsert",
        "columnDelimiter": "COMMA",
        "lineEnding": "LF",
        "numberRecordsFailed": 0,
        "numberRecordsProcessed": 0,
        "apexProcessingTime": 0,
        "apiActiveProcessingTime": 0,
        "apiVersion": 59,
        "concurrencyMode": "Parallel",
        "contentType": "CSV",
        "createdDate": "2024-03-19T23:30:28.000+0000",
        "jobType": "V2Ingest",
        "retries": 0,
        "systemModstamp": "2024-03-19T23:30:29.000+0000",
        "totalProcessingTime": 0
    },
    "message": "No records processed successfully. This is likely due the CSV being empty or issues with CSV column names."
}

```

Hubspot test

```
jaehyunlim@Jaehyuns-MacBook-Pro connectors % go run test/hubspot/write/main.go
{
  "success": true,
  "recordId": "5069569450",
  "data": {
    "company": "Numedii",
    "createdate": "2024-03-19T23:11:00.536Z",
    "email": "jalynbeatty@lubowitz.name",
    "firstname": "Monserrat",
    "hs_all_contact_vids": "5069569450",
    "hs_calculated_phone_number": "+16619403402",
    "hs_calculated_phone_number_country_code": "US",
    "hs_email_domain": "lubowitz.name",
    "hs_is_contact": "true",
    "hs_is_unworked": "true",
    "hs_lifecyclestage_lead_date": "2024-03-19T23:11:00.536Z",
    "hs_object_id": "5069569450",
    "hs_object_source": "INTEGRATION",
    "hs_object_source_id": "2317233",
    "hs_object_source_label": "INTEGRATION",
    "hs_pipeline": "contacts-lifecycle-pipeline",
    "hs_searchable_calculated_phone_number": "6619403402",
    "lastmodifieddate": "2024-03-19T23:11:00.536Z",
    "lastname": "Hermann",
    "lifecyclestage": "lead",
    "phone": "6619403402",
    "website": "http://www.nationalstrategic.com/empower/repurpose/distributed/web-readiness"
  }
}Wrote contact
Reading contacts..
{
  "rows": 0,
  "data": [],
  "done": true
}
```

### CODE REVIEW

**Comment by jlimatampersand** [comment]
Current dependencies on/for this PR:
* **#73** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/73?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/amp-labs/connectors/73?utm_source=stack-comment).

**Review by laurenzlong — COMMENTED** [review]
Can you please add a CONTRIBUTING.md file that explains how all of this works, so that someone who's adding a new test for a new provider will know what to do and how to run their test?

File: utils/credentialRegistry.go [hunk 1]
```diff
@@ -0,0 +1,442 @@
+// nolint: ireturn
```
Comment (jlimatampersand) [inline utils/credentialRegistry.go:1]: This file was mostly relocated. Just look at value reader portion.

File: test/hubspot/write.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"net/http"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/hubspot"
+	testUtils "github.com/amp-labs/connectors/test/utils"
+	"github.com/amp-labs/connectors/utils"
+)
+
+// GetHubspotConnector returns a Hubspot connector.
+func GetHubspotConnector(ctx context.Context, filePath string) *hubspot.Connector {
```
Comment (jlimatampersand) [inline test/hubspot/write.go:14]: Relocated from `test/utils/hubspot.go` some parts modified.

File: test/hubspot/write.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	registry := utils.NewCredentialsRegistry()
+
+	readers := []utils.Reader{
```
Comment (laurenzlong) [inline test/hubspot/write.go:17]: why doesn't the Hubspot test also use  AmpersandConnectionSchemaReaders?

File: test/utils/creds.go [hunk 4]
```diff
@@ -1,23 +1,69 @@
...
+	"github.com/amp-labs/connectors/utils"
 )
 
-// Credentials returns the credentials from the creds.json file.
-func Credentials(defaultFileName string) (*ajson.Node, error) {
-	fileName := defaultFileName
-
-	if fn, ok := os.LookupEnv("CREDENTIALS_FILE"); ok {
-		fileName = fn
+func AmpersandConnectionSchemaReaders(filePath string) []utils.Reader {
```
[the remark this answers is not in the document]
Comment (laurenzlong) [reply test/utils/creds.go:9]: Also the function name is a little weird, maybe just `JSONFileReaders`

File: test/utils/creds.go [hunk 5]
```diff
@@ -1,23 +1,69 @@
...
 
-	byteValue, err := os.ReadFile(fileName)
-	if err != nil {
-		return nil, err
-	}
+	return schema
+}
 
-	return ajson.Unmarshal(byteValue)
+func EnvVarSchemaReaders(prefix string) []utils.Reader {
```
Comment (laurenzlong) [inline test/utils/creds.go:46]: the word 'Schema' is not really necessary here, `EnvVarsReaders` would be a shorter function name.

**Review by laurenzlong — APPROVED** [review]

**Pull request merged**